### PR TITLE
Use tree height instead of chunk limit for merkleizer

### PR DESCRIPTION
**Pull request by arnetheduck**
More closely matches implementation and avoids generic bloat.

Renamed to avoid accidental limit-vs-height confusion.

### CODE REVIEW

File: ssz_serialization/merkleization.nim [hunk 1]
```diff
@@ -65,17 +65,26 @@ type
...
+    combinedChunks: array[height, (Digest, Digest)]
     totalChunks*: uint64 # Public for historical reasons
     topIndex: int
     internal: bool
+      # Avoid copying chunk data into merkleizer when not needed - maw result
+      # in an incomplete root-to-leaf proof
 
-template getChunkCount*(m: SszMerkleizer): uint64 =
+template limit*(T: type SszMerkleizer2): Limit =
+  if T.height == 0: 0'i64 else: 1'i64 shl (T.height - 1)
```
Comment (zah) [inline ssz_serialization/merkleization.nim:79]: Hrm, what about the cases where the limit is not a power of 2? We have to review more carefully how this code is being used.

Comment (arnetheduck) [reply ssz_serialization/merkleization.nim:79]: no critical usages, ie it doesn't materially matter since all trees are extended to power-of-2 anyway

**Comment by tersec** [comment]
https://github.com/nim-lang/Nim/pull/23188 should fix the CI issue in `devel` which this triggered.

**Comment by arnetheduck** [comment]
@tersec indeed it did: https://github.com/status-im/nim-ssz-serialization/actions/runs/7490848447 thanks!

**Pull request merged**